docs: Prep 3.3.x documentation (#97)
brown-ccv/honeycomb-docs: versioned_docs/version-3.2.x/project_organization/directory_structure.mdx

---
id: directory_structure
slug: /directory_structure
title: Directory Structure
description: Overview of Honeycomb's directory structure
---

> This project directory is organized to be very modular and composable. In general, files and functions should be relatively small and self-contained, global scope should not be used and only the pieces of code needed for any given file should be imported. This keeps the code maintainable with clear lineage and purpose for each piece of code.

Reviewer: ```suggestion
This project directory is organized to be very modular and composable. In general, files and functions should be relatively small and self-contained, global scope should not be used, and only the pieces of code needed for any given file should be imported. This keeps the code maintainable with clear lineage and purpose for each piece of code.
```

:::danger
The build folder is in Honeycomb's `.gitignore` and should never be added to git
:::

## emulator_data/

This folder contains starter data for the Firebase Emulators to use while developing locally. See the [Firebase Scripts](npm_scripts#firebase) to use the data.

:::danger
`emulator_data/` is written to when running `npm run firebase:emulators:save` and should never be manually edited.
:::

## env/

This folder contains different files used to pass environment variables (settings) into Honeycomb. Honeycomb starts with presets for common use cases and is explained in greater detail in the [Environment Variables](environment_variables) section.

## node_modules/

:::danger
`node_modules/` is written to when running `npm install` and should never be manually edited.
:::

## psiturkit/

The file `psiturk-it` inside `psiturkit/` is a bash script used to instal PsiTurk locally - see [PsiTurk](psiturk#build-instructions) for more information.

:::caution
Nothing in this folder should ever need to be manually edited.
:::

## public/

The `public` directory contains files that are used as assets in the built app.

> - `index.html` is the entry point of the website

Reviewer: Get rid of periods for these bullets?

  - Changing `<title>Honeycomb</title>` will update the name you can see in the browser tab.
- `favicon.ico` is the small icon you can see in the browser tab.
- `electron.js` contains all of the code related to the electron app.

### assets/

The `public/assets/` directory contains all of the images and videos needed to run your task.

### lib/

The `public/lib/` directory contains the files PsiTurk needs to run. Note that `index.html` references these files inside the `<script>` tags.

:::danger
These files are minified versions of dependencies and should never be manually edited.
:::

### config/

The `public/config/` directory contains the config files needed for the electron app. This includes the event-marker details and event codes. `version.json` keeps track of the current git commit, which Honeycomb uses to keep track of the version of your task used for a given experiment.

## src/

This folder contains the source code for the Honeycomb application.

### App/

Files relating to the React application. This is the code that runs your JsPsych task and shouldn't need to be changed.

#### components/

The React components that make up Honeycomb are located here.

> - `App.jsx` initializes and maintains the state of the application. It is also where communication is set up between the `electron` and `psiturk` processes.

Reviewer: Same, the first two have periods but last two don't, maybe just drop them? I won't add more comments about it, but check the files below this point and decide. In general, I think anything rendered as a list of bullets doesn't need periods, but any text written as sentences otherwise does.

- `Error.jsx` displays a small error message. It is rendered when the App.jsx detects an issue in state.
- `JsPsychExperiment.jsx` initializes the JsPsych experiment
- `Login.jsx` handles user authentication based on the environment variables passed to Honeycomb

#### deployments/

Custom code used by the various deployments such as Firebase.

:::caution
Changes to these files will change how Honeycomb handles data and should be done with great caution.
:::

### config/

Each file in the config directory contains settings for a different part of the task.

- `main.js` contains the global settings (e.g. whether Honeycomb is running online or in the clinic) passed from env variables and logic for loading the appropriate language file.

> - `trigger.js` for equipment-related settings (e.g. event markers). It uses a slightly different style of javascript as it is imported both in the React app as well as the electron process.

Reviewer: ```suggestion
- `trigger.js` for equipment-related settings (e.g. event markers). It uses a slightly different style of JavaScript as it is imported both in the React app as well as the electron process.
```

- `config.json` contains the settings for your task. Usage of the config file allows for easy updating of task settings. Common settings can be written once in the config file and re-used throughout the task.

> - `language.json` contains the language used in your task. Usage of language json files allows for easy internationalization of the task (e.g. english and spanish) and mturk-specific language. Common phrases can be written once in a language file and re-used throughout the task.

Reviewer: ```suggestion
- `language.json` contains the language used in your task. Usage of language JSON files allows for easy internationalization of the task (e.g. english and spanish) and mturk-specific language. Common phrases can be written once in a language file and re-used throughout the task.
```

### lib/

A library of utility and markup are located here. This allows for functions and html to be re-used wherever needed.

#### markup/

`src/lib/markup/` files contain HTML templates used throughout the task.

- `stimuli.js` contains a `baseStimulus` function that wraps some markup in a container that takes up 100% of the height and width of the viewport

> - `photodiode.js` contains the markup for the photodiode box and spot. It is displayed in the bottom right corner of the scree.

Reviewer: ```suggestion
- `photodiode.js` contains the markup for the photodiode box and spot, displayed in the bottom right corner of the screen
```
